### PR TITLE
Update to doctrine/coding-standard 10 on 4.0.x

### PR DESCRIPTION
**Pull request by greg0ire**
To be followed by a `git merge origin/3.5.x --strategy ours`

### CODE REVIEW

File: src/Cache/QueryCacheProfile.php [hunk 1]
```diff
@@ -22,12 +22,12 @@ class QueryCacheProfile
 {
     public function __construct(
         private readonly int $lifetime = 0,
-        private readonly ?string $cacheKey = null,
-        private readonly ?CacheItemPoolInterface $resultCache = null
+        private readonly string|null $cacheKey = null,
+        private readonly CacheItemPoolInterface|null $resultCache = null,
```
Comment (morozov) [inline src/Cache/QueryCacheProfile.php]: I don't believe that union types should be enforced over the nullable ones. We discussed it recently in another thread.

Comment (greg0ire) [reply src/Cache/QueryCacheProfile.php]: Can you please open an issue on the CS repo? I think that's were the issue needs to be discussed.

**Comment by greg0ire** [comment]
Psalm suddenly chokes on `Connection&MockObject`

Psalm 5.0.0-beta1 doesn't have the issue but comes with other errors…

File: src/Cache/QueryCacheProfile.php (hunk 1 above)
Comment (morozov) [reply src/Cache/QueryCacheProfile.php]: Done: https://github.com/doctrine/coding-standard/issues/288.

**Review by morozov — CHANGES_REQUESTED** [review]
See https://github.com/doctrine/dbal/pull/5632#discussion_r957742275.

**Comment by greg0ire** [comment]
I'll revert the change about union types that way we don't have to wait for the resolution of https://github.com/doctrine/coding-standard/issues/288 to  merge this.

**Review by morozov — APPROVED** [review]

**Pull request merged**